src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java

```diff
@@ -17,138 +18,150 @@
-  }
-
-  public void typeInSearchField(String content) {
-    wait.forElementVisible(searchInput);
-    searchInput.sendKeys(content);
-  }
-
-  public boolean isUserLoggedIn(String username) {
-    return driver.findElements(By.cssSelector("figure.avatar img[alt='" + username + "']")).size()
-           > 0;
-  }
+    @FindBy(css = ".ember-text-field")
+    private WebElement searchInput;
+    @FindBy(css = ".cancel")
+    private WebElement cancelSearchCaption;
+    @FindBy(css = ".local-wikia-search a")
+    private WebElement searchSuggestion;
+    @FindBy(css = ".local-nav-menu > li")
+    private List<WebElement> navList;
+    @FindBy(css = ".back")
+    private WebElement backChevron;
+    @FindBy(css = ".random-article-link")
+    private WebElement randomPageButton;
+    @FindBy(css = ".overlay")
+    private WebElement overlay;
+    @FindBy(css = ".local-wikia-search")
+    private WebElement resultField;
+    @FindBy(css = ".local-nav-menu")
+    private WebElement menuField;
+    @FindBy(css = ".nav")
+    private WebElement searchButton;
+    @FindBy(css = "nav.side-nav")
+    private WebElement menuView;
+    @FindBy(xpath = "//span[contains(.,'Sorry, we could')]")
+    private WebElement sorrySpan;
+    @FindBy(css = "div.side-nav-toggle")
+    private WebElement openListIcon;
+
+
+    public NavigationSideComponentObject(WebDriver driver) {
+        super(driver);
+    }
+
+    public void clickSearchField() {
+        wait.forElementVisible(searchInput);
+        searchInput.click();
+    }
+
+    public void clickSearchButton() {
+        wait.forElementVisible(searchButton);
+        searchButton.click();
+    }
+
+    public void clickBackChevron() {
+        wait.forElementVisible(backChevron);
+        backChevron.click();
+    }
+
+    public void clickCancelButton() {
+        wait.forElementVisible(cancelSearchCaption);
+        cancelSearchCaption.click();
+    }
+
+    public void clickSuggestion(int index) {
+        wait.forElementVisible(searchSuggestion);
+        searchSuggestion.click();
+    }
+
+    public void clickNavListElement(int index) {
+        wait.forElementVisible(navList.get(index));
+        navList.get(index).click();
+    }
+
+    public void clickOverlay() {
+        wait.forElementVisible(overlay);
+        tapOnElement(overlay);
+    }
+
+    public ArticlePageObject clickRandomPageButton() {
+        wait.forElementVisible(randomPageButton);
+        randomPageButton.click();
+
+        return new ArticlePageObject(driver);
+    }
+
+    public boolean isSuggestionListDisplayed() {
+        try {
+            wait.forElementVisible(searchSuggestion, 5, 1000);
+        } catch (TimeoutException | NoSuchElementException e) {
+            return false;
+        }
+        return true;
+    }
+
+    public boolean isSorryInfoDisplayed() {
+        try {
+            wait.forElementVisible(sorrySpan, 5, 1000);
+        } catch (TimeoutException | NoSuchElementException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)